### ISSUE TITLE
IEx crash on remote shell

### ISSUE THREAD

**Issue by Gazler** [elixir-lang/elixir#3889]
I am trying to run iex --remsh from my local machine:

    gazler@gazler-desktop-pc:~$ iex --remsh remote@remotenode --cookie hunter2 --name "local@localnode"
    Erlang/OTP 18 [erts-7.1] [source-2882b0c] [64-bit] [smp:8:8] [async-threads:10] [hipe] [kernel-poll:false]

    *** ERROR: Shell process terminated! (^G to start new job) ***

My versions locally:

    Erlang/OTP 18 [erts-7.1] [source-2882b0c] [64-bit] [smp:4:4] [async-threads:10] [hipe] [kernel-poll:false]

    Interactive Elixir (1.1.0) - press Ctrl+C to exit (type h() ENTER for help)

And on the server:

    Erlang/OTP 18 [erts-7.1] [source-2882b0c] [64-bit] [smp:4:4] [async-threads:10] [hipe] [kernel-poll:false]

    Interactive Elixir (1.1.0) - press Ctrl+C to exit (type h() ENTER for help)

Connecting with Erlang seemed to work:

    erl -remsh remote@remotenode"" -setcookie hunter2 -name "local@localnode"

/cc @josevalim

**Comment by eraserewind** [comment]
I'm actually having the same problem on a Nerves node. I guessed it was a build problem with nerves or something:

```
$ iex --sname creature --cookie {redacted} --remsh gsm_relay@nerves-3f77
Erlang/OTP 18 [erts-7.1] [source] [64-bit] [smp:4:4] [ds:4:4:10] [async-threads:10] [hipe] [kernel-poll:false]

*** ERROR: Shell process terminated! (^G to start new job) ***
```

However when using the `^G` menu, the connection works fine. Could you try this @Gazler ?

```
$ iex --sname creature --cookie {redacted}
Erlang/OTP 18 [erts-7.1] [source] [64-bit] [smp:4:4] [ds:4:4:10] [async-threads:10] [hipe] [kernel-poll:false]

Interactive Elixir (1.1.1) - press Ctrl+C to exit (type h() ENTER for help)
iex(creature@creature)1>
User switch command
 --> r 'gsm_relay@nerves-3f77' 'Elixir.IEx'
 --> c
Interactive Elixir (1.1.1) - press Ctrl+C to exit (type h() ENTER for help)
iex(gsm_relay@nerves-3f77)1>
```

**Comment by Gazler** [comment]
@eraserewind that definitely works.

**Comment by fishcakez** [comment]
Fixed in 361d2f065d53e1dab2441f100d90841cd0fc7965

**Comment by josevalim** [comment]
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:
